src/phases/move-effect-phase.ts

```diff
@@ -192,7 +192,9 @@ export class MoveEffectPhase extends PokemonPhase {
              * made visible to the user until the resulting {@linkcode DamagePhase}
              * is invoked.
              */
-          const hitResult = !isProtected ? target.apply(user, move) : HitResult.NO_EFFECT;
+          const moveCategory = new Utils.IntegerHolder(move.category);
```

Comment:
```suggestion
          const moveCategory = new Utils.NumberHolder(move.category);
```
I think we're trying to move away from using integer since there's no actual difference with number

These lines may need to be updated too, but i think that's it?
https://github.com/pagefaultgames/pokerogue/blob/d8c914c7686c0fa4c74609ebed97c03f65a46f3e/src/data/move.ts#L4110-L4114

Reply:
Ok, fixed!